crates/holochain/src/conductor/interface.rs

```diff
@@ -1,145 +1,36 @@
-use crate::conductor::api::*;
-use error::InterfaceResult;
-use futures::{
-    channel::mpsc::{channel, Receiver, Sender},
-    future::{BoxFuture, FutureExt},
-    sink::{Sink, SinkExt},
-    stream::{Stream, StreamExt},
-};
-use holochain_serialized_bytes::{SerializedBytes, SerializedBytesError};
+use crate::{conductor::api::*, core::signal::Signal};
+use error::{InterfaceError, InterfaceResult};
 use serde::{Deserialize, Serialize};
-use std::convert::{TryFrom, TryInto};
+use std::convert::TryInto;
+use tokio::sync::broadcast;
 
 pub mod error;
 pub mod websocket;
 
-/// Allows the conductor or cell to forward signals to connected clients
-pub struct ConductorSideSignalSender<Sig>
-where
-    Sig: 'static + Send + TryInto<SerializedBytes, Error = SerializedBytesError>,
-{
-    sender: Sender<SerializedBytes>,
-    phantom: std::marker::PhantomData<Sig>,
-}
+#[derive(Clone, Debug)]
+pub struct SignalBroadcaster(Vec<broadcast::Sender<Signal>>);
 
-impl<Sig> ConductorSideSignalSender<Sig>
-where
-    Sig: 'static + Send + TryInto<SerializedBytes, Error = SerializedBytesError>,
-{
+impl SignalBroadcaster {
     /// send the signal to the connected client
-    pub async fn send(&mut self, data: Sig) -> InterfaceResult<()> {
-        let data: SerializedBytes = data.try_into()?;
-        self.sender.send(data).await?;
+    pub async fn send(&mut self, sig: Signal) -> InterfaceResult<()> {
```

Comment:
There's nothing async about this fn so the async should be removed so down further we don't need the tokio block on. That's the best part about broadcast Sender's is they don't block :slightly_smiling_face: